lib/nerve.rb

```diff
@@ -61,107 +69,118 @@ def load_config!
-              log.info "nerve: launching new watchers: #{services_to_launch}"
-              services_to_launch.each do |name|
-                launch_watcher(name, @watchers_desired[name])
+              unless services_to_launch.empty?
+                log.info "nerve: launching new watchers: #{services_to_launch}"
+                services_to_launch.each do |name|
+                  launch_watcher(name, @watchers_desired[name])
+                end
               end
-            end
 
-            # Detect and update existing service watchers which are in both
-            # the currently running state and the desired (config) watcher
-            # state but have different configurations
-            services_to_update = @watchers.select { |name, _|
-              @watchers_desired.has_key?(name) &&
-              merged_config(@watchers_desired[name], name).hash != @watcher_versions[name]
-            }.keys()
-
-            services_to_update.each do |name|
-              log.info "nerve: detected new config for #{name}"
-              # Keep the old watcher running until the replacement is launched
-              # This keeps the service registered while we change it over
-              # This also keeps connection pools active across diffs
-              temp_name = "#{name}_#{@watcher_versions[name]}"
-              @watchers[temp_name] = @watchers.delete(name)
-              @watcher_versions[temp_name] = @watcher_versions.delete(name)
-              log.info "nerve: launching new watcher for #{name}"
-              launch_watcher(name, @watchers_desired[name], :wait => true)
-              log.info "nerve: reaping old watcher #{temp_name}"
-              reap_watcher(temp_name)
+              # Detect and update existing service watchers which are in both
+              # the currently running state and the desired (config) watcher
+              # state but have different configurations
+              services_to_update = @watchers.select { |name, _|
+                @watchers_desired.has_key?(name) &&
+                merged_config(@watchers_desired[name], name).hash != @watcher_versions[name]
+              }.keys()
+
+              services_to_update.each do |name|
+                log.info "nerve: detected new config for #{name}"
+                # Keep the old watcher running until the replacement is launched
+                # This keeps the service registered while we change it over
+                # This also keeps connection pools active across diffs
+                temp_name = "#{name}_#{@watcher_versions[name]}"
+                @watchers[temp_name] = @watchers.delete(name)
+                @watcher_versions[temp_name] = @watcher_versions.delete(name)
+                log.info "nerve: launching new watcher for #{name}"
+                launch_watcher(name, @watchers_desired[name], :wait => true)
+                log.info "nerve: reaping old watcher #{temp_name}"
+                reap_watcher(temp_name)
```

Comment:
the main operations in the main loop is to launch_watcher and reap_watcher for different cases, new/update/revive. Would it make sense to do a statsd counter at the beginning of the launch_watcher an reap_watcher functions, so that we can capture all of them?